Bug: https://trac.sagemath.org/ticket/29719

From a3981f6eb02a3d9e7a78aa5c81525fbd0e3fbc96 Mon Sep 17 00:00:00 2001
From: Dima Pasechnik <dima@pasechnik.info>
Date: Sat, 23 May 2020 10:04:26 +0100
Subject: skip square-freeness test for polynomials non-fields

Due to change in the way Flint deals with this case,
which now throws an exception rather than returns 1 regardless,
we explicitly check for field-ness in the calls to
HyperellipticCurve
---
 src/sage/schemes/hyperelliptic_curves/monsky_washnitzer.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

From 0d5a2e6b53dca951fb225d38ef9abc45ce264b62 Mon Sep 17 00:00:00 2001
From: Dima Pasechnik <dima@pasechnik.info>
Date: Sat, 23 May 2020 10:52:07 +0100
Subject: fix and make robust related doctests

flint 2.6.0 (stable)
---
 build/pkgs/flint/checksums.ini                  |  8 ++++----
 src/doc/en/constructions/algebraic_geometry.rst | 14 +++++++++-----
 src/doc/en/developer/coding_in_other.rst        |  9 +--------
 src/sage/modules/fg_pid/fgp_module.py           |  7 ++++---
 src/sage/modules/free_module_morphism.py        | 12 +++++++-----
 5 files changed, 25 insertions(+), 25 deletions(-)



--- a/sage/src/sage/schemes/hyperelliptic_curves/monsky_washnitzer.py
+++ b/sage/src/sage/schemes/hyperelliptic_curves/monsky_washnitzer.py
@@ -1876,9 +1876,9 @@
             if not hasattr(self, '_curve'):
                 if self._Q.degree() == 3:
                     ainvs = [0, self._Q[2], 0, self._Q[1], self._Q[0]]
-                    self._curve = EllipticCurve(ainvs)
+                    self._curve = EllipticCurve(ainvs, check_squarefree=R.is_field())
                 else:
-                    self._curve = HyperellipticCurve(self._Q)
+                    self._curve = HyperellipticCurve(self._Q, check_squarefree=R.is_field())
 
         else:
             raise NotImplementedError("Must be an elliptic curve or polynomial "
--- a/sage/src/doc/en/constructions/algebraic_geometry.rst
+++ b/sage/src/doc/en/constructions/algebraic_geometry.rst
@@ -142,13 +142,17 @@
        sage: I = singular.ideal('[x^4+x, y^4+y]')
        sage: L = singular.closed_points(I)
        sage: # Here you have all the points :
-       sage: print(L)
+       sage: L       # random
        [1]:
-          _[1]=y+1  # 32-bit
-          _[2]=x+1  # 32-bit
-          _[1]=y    # 64-bit
-          _[2]=x    # 64-bit
+          _[1]=y+1
+          _[2]=x+1
        ...
+       sage: l=[L[k].sage() for k in [1..10]]; len(l) # there are 10 points
+       10
+       sage: r=sorted(l[0].ring().gens()); r
+       [y, x]
+       sage: r in [t.gens() for t in l] #  one of them is given by [y,x]
+       True
 
 -  Another way to compute rational points is to use Singular's
    ``NSplaces`` command. Here's the Klein quartic over :math:`GF(8)`
--- a/sage/src/doc/en/developer/coding_in_other.rst
+++ b/sage/src/doc/en/developer/coding_in_other.rst
@@ -439,7 +439,7 @@
     ''
     sage: L = singular.eval("POINTS;")
 
-    sage: print(L)
+    sage: print(L) # random
     [1]:
        [1]:
           0
@@ -447,13 +447,6 @@
           1
        [3]:
           0
-    [2]:
-       [1]:
-          -2
-       [2]:
-          -1
-       [3]:
-          1
     ...
 
 From looking at the output, notice that our wrapper function will need
--- a/sage/src/sage/modules/fg_pid/fgp_module.py
+++ b/sage/src/sage/modules/fg_pid/fgp_module.py
@@ -127,7 +127,8 @@
 Here we illustrate lifting an element of the image of f, i.e., finding
 an element of M0 that maps to a given element of M1::
 
-    sage: y = f.lift(3*M1.0); y
+    sage: y = f.lift(3*M1.0)
+    sage: y # random
     (0, 13)
     sage: f(y)
     (3)
@@ -1285,7 +1286,7 @@
             (0, 4)
             sage: Q.coordinate_vector(x, reduce=True)
             (0, 4)
-            sage: Q.coordinate_vector(-x, reduce=False)
+            sage: Q.coordinate_vector(-x, reduce=False) # random
             (0, -4)
             sage: x == 4*Q.1
             True
@@ -1414,7 +1415,7 @@
             Echelon basis matrix:
             [ 0 12  0]
             [ 0  0  4]
-            sage: X
+            sage: X         # random
             [0 4 0]
             [0 1 0]
             [0 0 1]
--- a/sage/src/sage/modules/free_module_morphism.py
+++ b/sage/src/sage/modules/free_module_morphism.py
@@ -350,12 +350,14 @@
             sage: V = X.span([[2, 0], [0, 8]], ZZ)
             sage: W = (QQ**1).span([[1/12]], ZZ)
             sage: f = V.hom([W([1/3]), W([1/2])], W)
-            sage: f.lift([1/3])
+            sage: l=f.lift([1/3]); l # random
             (8, -16)
-            sage: f.lift([1/2])
-            (12, -24)
-            sage: f.lift([1/6])
-            (4, -8)
+            sage: f(l)
+            (1/3)
+            sage: f(f.lift([1/2]))
+            (1/2)
+            sage: f(f.lift([1/6]))
+            (1/6)
             sage: f.lift([1/12])
             Traceback (most recent call last):
             ...
